src/transport/SessionManager.cpp

```diff
@@ -568,11 +568,21 @@ void SessionManager::OnMessageReceived(const PeerAddress & peerAddress, System::
     }
 }
 
-void SessionManager::UnauthenticatedMessageDispatch(const PacketHeader & packetHeader, const Transport::PeerAddress & peerAddress,
+void SessionManager::UnauthenticatedMessageDispatch(PacketHeader & packetHeader, const Transport::PeerAddress & peerAddress,
```

Comment:
This "pass a non-const-ref" bit makes it look like the caller cares about the result, but in practice it doesn't, right?

How about:

1. We make this a const ref, but rename to "partialPacketHeader" or some such.
2. The Unicast/Insecure callees then instantiate another PacketHeader called `packetHeader` and decode into that.
3. The group callee is already instantiating another PacketHeader.

Thoughts?  I would be OK with a followup for these changes, because we'd want to see the impact it has on the stack and whatnot.

Reply:
Reverted back to const and renamed `partialPacketHeader` as suggested.